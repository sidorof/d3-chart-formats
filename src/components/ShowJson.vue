<template>
  <div class="text-center">
    <v-dialog
      v-model="dialog"
      width="500"
    >
      <template v-slot:activator="{ on }">
        <v-btn
          color="red lighten-2"
          dark
          v-on="on"
        >
          Show JSON
        </v-btn>
      </template>

      <v-card>
        <v-card-title
          class="headline grey lighten-2"
          primary-title
        >
          JSON Parameters sent to backend
        </v-card-title>

        <v-card-text>
          {{ paramsFormatted }}
        </v-card-text>

        <v-divider></v-divider>

        <v-card-actions>
          <div class="flex-grow-1"></div>
          <v-btn
            color="primary"
            text
            @click="dialog = false"
          >
            Done
          </v-btn>
        </v-card-actions>
      </v-card>
    </v-dialog>
  </div>
</template>

<script>
export default {
  props: ['chartParams'],
  data () {
    return {
      dialog: false
    }
  },
  computed: {
    paramsFormatted () {
      return JSON.stringify(this.chartParams, null, '\t')
    }
  }
}

</script>
