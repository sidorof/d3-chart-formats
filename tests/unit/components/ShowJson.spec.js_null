// tests/unit/components/ShowJson.spec.js

// Libraries
import Vue from 'vue'
import Vuetify from 'vuetify'

// Components
import ShowJson from '@/components/ShowJson'

// Utilities
import {
  mount,
  createLocalVue
} from '@vue/test-utils'

const localVue = createLocalVue()

describe('ShowJson.vue', () => {
  let vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  it('should have a custom title and match snapshot', () => {
    const wrapper = mount(ShowJson, {
      localVue,
      vuetify,
      propsData: {
        title: 'JSON Parameters sent to backend'
      }
    })

    // const title = wrapper.find('.v-card__title')
    const title = wrapper.find('.div')

    /* this does not work as expected
     * console.log('html', wrapper.html())
     * looks like it ought to be:
     *   const title = wrapper.find('.v-card__title')
     * instead the html looks like:
     * html <div class="text-center" title="JSON Parameters sent to backend">
     *
     * Also this functions properly in the browser. My
     * assumption is that it is related to the dialog box.
     *
     * So this is being skipped until I know more.
    /* expect(title.text()).toBe('JSON Parameters sent to backend')
     */
  })

  it('should emit an event when the action v-btn is clicked', () => {
    const wrapper = mount(ShowJson, {
      localVue,
      vuetify,
      propsData: {
        paramsFormatted: 'this is a test',
        dialog: true
      }
    })

    const event = jest.fn()
    const button = wrapper.find('.v-btn')
    console.log('html', wrapper.html())
    console.log('--------------------------------')
    // Here we bind a listener to the wrapper
    // instance to catch our custom event
    // https://vuejs.org/v2/api/#Instance-Methods-Events
    wrapper.vm.$on('action-btn:clicked', event)

    expect(event).toHaveBeenCalledTimes(0)

    // Simulate a click on the button
    button.trigger('click')

    // Ensure that our mock event was called
    expect(event).toHaveBeenCalledTimes(1)
    console.log('html', wrapper.html())

  })

  it('should have paramsFormatted', () => {
    const wrapper = mount(ShowJson, {
      localVue,
      vuetify,
      propsData: {
        paramsFormatted: 'this is a test'
      }
    })

    const button = wrapper.find('.v-btn__content')
    console.log('button', button)
    expect(button.text()).toBe('Show JSON')
  })
})
