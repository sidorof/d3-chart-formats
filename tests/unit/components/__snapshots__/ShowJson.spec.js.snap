// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowJson.vue should have a custom title and match snapshot 1`] = `
<div class="text-center" title="JSON Parameters sent to backend">
  <v-dialog width="500">
    <v-card>
      <v-card-title primary-title="" class="headline grey lighten-2">
        JSON Parameters sent to backend
      </v-card-title>
      <v-card-text>

      </v-card-text>
      <v-divider></v-divider>
      <v-card-actions>
        <div class="flex-grow-1"></div>
        <v-btn color="primary" text="">
          Done
        </v-btn>
      </v-card-actions>
    </v-card>
  </v-dialog>
</div>
`;
